lib/puppet_pal.rb

```diff
@@ -389,8 +362,30 @@ def initialize(task_type)
     # @return [Boolean] if the given arguments are acceptable when running the task
     #
     def runnable_with?(args_hash)
-      @new_signature ||= FunctionSignature.new(@task_type.new_function)
-      @new_signature.callable_with?(args_hash)
+      params = @task.parameters
+      params_type = if params.nil?
+        T_GENERIC_TASK_HASH
+      else
+        key_to_type = {}
+        @task.parameters.each_pair { |k, v| key_to_type[k] = v['type'] }
+        Puppet::Pops::Types::TypeFactory.struct(key_to_type)
+      end
+      params_type.instance?(args_hash)
+    end
```

Comment:
I forgot to add  the same support for calling a lambda with a type mismatch error message as in the callable_with? for functions and plans. The intention was that `runnable_with?` should do the same. Not sure if that alters the approach to type checking, and/or if that should be done here, or in a separate ticket.